### PR TITLE
Excluded Resources

### PR DESCRIPTION
**Pull request by Hbbb**
* Adds excluded resources to `model.TuberApp`
* Excludes resources in releaser

TODO:
* Implement UI/GraphQL to set/edit excluded resources

### CODE REVIEW

File: pkg/reviewapps/reviewapps.go [hunk 1]
```diff
@@ -91,16 +91,17 @@ func CreateReviewApp(ctx context.Context, db *core.DB, l *zap.Logger, branch str
...
+		ImageTag:          imageTag,
+		Name:              reviewAppName,
+		Paused:            false,
+		ReviewApp:         true,
+		SlackChannel:      sourceApp.SlackChannel,
+		SourceAppName:     sourceApp.Name,
+		State:             nil,
+		TriggerID:         triggerID,
+		Vars:              vars,
+		ExcludedResources: sourceApp.ReviewAppsConfig.ExcludedResources,
```
Comment (jkminneti) [inline pkg/reviewapps/reviewapps.go:104]:
```suggestion
		ExcludedResources: sourceApp.ExcludedResources,
```

File: graph/schema.resolvers.go [hunk 2]
```diff
@@ -233,3 +249,13 @@ func (r *Resolver) TuberApp() generated.TuberAppResolver { return &tuberAppResol
...
 type tuberAppResolver struct{ *Resolver }
+
+// !!! WARNING !!!
+// The code below was going to be deleted when updating resolvers. It has been copied here so you have
+// one last chance to move it out of harms way if you want. There are two reasons this happens:
+//  - When renaming or deleting a resolver the old code will be put in here. You can safely delete
+//    it when you're done.
+//  - You have helper methods in this file. Move them out to keep these resolver files clean.
+func (r *mutationResolver) ExcludedResources(ctx context.Context) ([]*model.Resource, error) {
+	panic(fmt.Errorf("not implemented"))
```
Comment (jkminneti) [inline graph/schema.resolvers.go:260]: gen problem or just actually unimplemented?

File: pkg/reviewapps/reviewapps.go (hunk 1 above)
Comment (jkminneti) [reply pkg/reviewapps/reviewapps.go:104]: or also just fixed it in https://github.com/Freshly/tuber/pull/208

**Review by quinn-freshly — CHANGES_REQUESTED** [review]
as per convo on friday can you change this to be on the app not the review app config

**Comment by jkminneti** [comment]
> as per convo on friday can you change this to be on the app not the review app config

he did tho

File: graph/schema.resolvers.go (hunk 2 above)
Comment (Hbbb) [reply graph/schema.resolvers.go:289]: It always gens as unimplemented like this. I haven't implemented it yet. I think I tagged you a bit too early, sorry.

File: pkg/core/releaser_test.go [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+				errorScope:       tt.fields.errorScope,
+				app:              tt.fields.app,
+				digest:           tt.fields.digest,
+				data:             tt.fields.data,
+				releaseYamls:     tt.fields.releaseYamls,
+				prereleaseYamls:  tt.fields.prereleaseYamls,
+				postreleaseYamls: tt.fields.postreleaseYamls,
+				db:               tt.fields.db,
+			}
+			if got := r.exclude(tt.args.res); !reflect.DeepEqual(got, tt.want) {
```
[the remark this answers is not in the document]
Comment (jkminneti) [reply pkg/core/releaser_test.go:100]: ooooops

[the remark this answers is not in the document]
Comment (jkminneti) [reply pkg/core/releaser_test.go:100]: uhhhh not sure what to do here lol i deleted the only thing that test was testing

[the remark this answers is not in the document]
Comment (jkminneti) [reply pkg/core/releaser_test.go:100]: we can revert if you want. i haven't only cus i kinda like my solution (where the _moment_ we have a kind and name, they get chucked)

**Review by quinn-freshly — APPROVED** [review]

**Pull request merged**